### PR TITLE
[CMSIS-NN] Include clip in the qnn binary op patterns

### PR DESCRIPTION
**Pull request by ashutosh-arm**
Support clip op in qnn binary op patterns for CMSIS-NN

### CODE REVIEW

**Comment by ashutosh-arm** [comment]
cc: @Mousius @manupa-arm @lhutton1

File: src/relay/backend/contrib/cmsisnn/relay_to_tir.cc [hunk 1]
```diff
@@ -548,7 +560,21 @@ class RelayToTIRVisitor : public MixedModeMutator {
...
 
   void EmitAdd(const GlobalVar& global_var, const Expr& expr) {
-    auto* add_call = expr.as<CallNode>();
+    const CallNode* clip_call = nullptr;
+    const CallNode* add_call = nullptr;
+    const CallNode* final_call = expr.as<CallNode>();
+    const OpNode* final_op = final_call->op.as<OpNode>();
+    int32_t output_min = std::numeric_limits<int8_t>::min();
+    int32_t output_max = std::numeric_limits<int8_t>::max();
+    if (final_op->name == "clip") {
```
Comment (manupa-arm) [inline src/relay/backend/contrib/cmsisnn/relay_to_tir.cc:569]: Can we refactor the parsing of binop --> clip into a function ?

As for a suggestion :
Something like : ParseBinaryElementwiseOpClipPattern(const Expr& expr) --> 
struct BinaryElementwiseClipPattern {
    Call binary_operator;
    Optional<Call> clip_operator;
}

So we can avoid duplicating the same code at two places.

**Review by manupa-arm — COMMENTED** [review]
A suggestion to remove duplication of code, otherwise LGTM.

File: src/relay/backend/contrib/cmsisnn/relay_to_tir.cc (hunk 1 above)
Comment (ashutosh-arm) [reply src/relay/backend/contrib/cmsisnn/relay_to_tir.cc]: Done. Thanks @manupa-arm. That was indeed needed.

**Review by manupa-arm — APPROVED** [review]
LGTM!

**Comment by manupa-arm** [comment]
Thanks @ashutosh-arm!

**Pull request merged**